orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/expressions/functions/DeployedServerGroupsExpressionFunctionProvider.java

```diff
@@ -83,13 +87,32 @@ public String getNamespace() {
             deployDetails.put("serverGroup", serverGroups.get(0));
           }
 
+          DeploymentContext deploymentContext = stage.mapTo(DeploymentContext.class);
+          List<Map<String, Object>> deployments = Optional.ofNullable(deploymentContext.tasks).orElse(emptyList()).stream()
+            .flatMap(task -> Optional.ofNullable(task.results).orElse(emptyList()).stream())
+            .flatMap(result -> Optional.ofNullable(result.deployments).orElse(emptyList()).stream())
+            .collect(Collectors.toList());
+          deployDetails.put("deployments", deployments);
```

Comment:
Just a question, why are you using optionals here? Does it provide any advantages over checking if something exists and only adding deployments if they exist?

Reply:
I suppose it is `Optional.ofNullable(foo).orElse(emptyList()).stream()` or `(foo == null ? emptyList() : foo).stream()`. Stylistic preference?

Reply:
I don't have a big preference. I think if statements are slightly more clear in general, but feel free to leave as is.